### PR TITLE
Unit Tests for the plugin using sbt-scripted

### PR DESCRIPTION
**Pull request by brkyvz**
@marmbrus 

These tests don't pass on master, but they will pass once PR #4 gets merged.
Some things below are huuuuuggeee hacks, I can explain them if you want to learn more the rationale.

### CODE REVIEW

**Comment by suhailshergill** [comment]
@marmbrus would be a good idea to enable travis-ci builds as well (adding a minimal `.travis.yml` along with a travis-ci badge to README.md would suffice)

File: sbt/sbt [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+# limitations under the License.
+#
+
+# When creating new tests for Spark SQL Hive, the HADOOP_CLASSPATH must contain the hive jars so
+# that we can run Hive to generate the golden answer.  This is not required for normal development
+# or testing.
+for i in "$HIVE_HOME"/lib/*
+do HADOOP_CLASSPATH="$HADOOP_CLASSPATH:$i"
+done
+export HADOOP_CLASSPATH
```
Comment (marmbrus) [inline sbt/sbt]: We can probably remove this hack.

**Comment by marmbrus** [comment]
I will let @ahirreddy do a more thorough review, but I have some high level feedback.  I like the mocked unit tests, but I wonder if (long term) we want some integration tests too.  If we add a travis build, one option would be to use real credentials that are [encrypted](http://docs.travis-ci.com/user/encryption-keys/).

File: sbt/sbt (hunk 1 above)
Comment (ahirreddy) [inline sbt/sbt]: +1

File: src/sbt-test/sbt-databricks/rest-calls/project/plugins.sbt [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+  libraryDependencies += "org.mockito" % "mockito-all" % "1.9.5"
+
+  // libraryDependencies += "org.apache.httpcomponents" % "httpclient" % "4.3.3"
```
Comment (ahirreddy) [inline src/sbt-test/sbt-databricks/rest-calls/project/plugins.sbt]: Remove this line

**Comment by ahirreddy** [comment]
LGTM

**Comment by brkyvz** [comment]
I'll follow up on these comments in the next PR. Thanks @ahirreddy !

**Pull request merged**